### PR TITLE
Replace chart-js with backend rendering

### PR DESCRIPTION
**Pull request by weeman1337**
ToDo

- [x] print styles
- [x] twigify

<details>
  <summary>Screenshots</summary>

chart5

chart10

chart15

chart20

chart25

chart30

Ugly-mode :wink: 

chart_uglymode

</details>

### CODE REVIEW

File: includes/helper/graph_helper.php [hunk 1]
```diff
@@ -17,30 +43,88 @@ function bargraph($dom_id, $key, $row_names, $colors, $data)
...
-        foreach ($data as $dataset) {
-            $values[] = $dataset[$row_key];
+    $group_labels = [];
+    $max = 0;
+
+    foreach ($data as $group_key => $group_data) {
+        $date = DateTimeImmutable::createFromFormat('Y-m-d', $group_key);
+        $group_labels[$group_key] = $group_key;
+
+        if ($date !== false) {
```
Comment (MyIgel) [inline includes/helper/graph_helper.php:53]: Imho that is redundant?

**Review by MyIgel — CHANGES_REQUESTED** [review]
I am not that convinced if introducing a new css naming convention (`.barchart__[...]`) which differs from the bootstrap names would be a great idea as it breaks consistency with bootstrap -> @marudor do you have an opinion on this?

It would also be nice to render a twig template for the charts :angel:

File: resources/assets/themes/bar-chart.scss [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    --barchart-bar-margin: .5px;
+    --barchart-group-margin: 0.2%;
+
+    .barchart__group:nth-child(4n) .barchart__xlabel {
+        // hide every second label
+        display: none;
+    }
+}
+
+.barchar__graph-container {
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply resources/assets/themes/bar-chart.scss:36]: :bearded_person: chart? :trollface:

**Comment by marudor** [comment]
Honestly, as long as its ensured that CSS classNames do not clash its fine to have a different naming pattern.
Afterall it's mainly personal choice.

**Comment by weeman1337** [comment]
@MyIgel 

- CSS classes conform now to the Igel-style :trollface: 
- Less DOM nodes
- :x: `includes` → Twig + Helper class
- Print view works

File: src/Helpers/BarChart.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Helpers;
+
+use DateTimeImmutable;
+
+class BarChart
```
Comment (MyIgel) [inline src/Helpers/BarChart.php:9]: I'm missing some tests for that class ;)

File: resources/views/components/barchart.twig [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                {% endfor %}
+                <div class="barchart-xlabel">{{ group.label }}</div>
+            </div>
+        {% endfor %}
+    </div>
+</div>
+
+<div class="barchart-legend">
+    {% for key, color in colors %}
+        <div
```
Comment (MyIgel) [inline resources/views/components/barchart.twig:26]: Another space? 0o

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: src/Helpers/BarChart.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                $groupLabels[$groupKey] = $date->format(__('Y-m-d'));
+            }
+
+            foreach ($rowLabels as $rowKey => $rowName) {
+                $max = max($max, $groupData[$rowKey]);
+            }
+        }
+
+        $roundedMax = (int) ceil($max / 5) * 5;
+        $step = $roundedMax / 5;
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/Helpers/BarChart.php:40]: We can use stylelint :D

File: resources/assets/themes/theme0.scss [hunk 6]
```diff
@@ -14,3 +14,4 @@ h1, h2, h3, h4, h5, h6, .h1, .h2, .h3, .h4, .h5, .h6 {
       color: #fff;
   }
 }
+
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply resources/assets/themes/theme0.scss:17]: We need stylelint :D

**Pull request merged**